apps/ide/src/plugins/webida.editor.example.multi-viewer/MultiViewerExampleEditorPart.js

```diff
@@ -217,14 +217,17 @@ define([
                     formViewer.createAdapter(parentNode);
                     that.initFormViewer();
                     formViewer.setContents(doc);
-                });
 
-                //3. codeViewer
-                var codeViewer = that.getCodeViewer();
-                that.addViewer('CodeEditor', 'Code Editor', codeViewer, 1, function(parentNode) {
-                    codeViewer.setParentNode(parentNode);
-                    that.initCodeEditor();
-                    codeViewer.setContents(doc);
+                    //3. codeViewer
+                    var codeViewer = that.getCodeViewer();
+                    that.addViewer('CodeEditor', 'Code Editor', codeViewer, 1, function(parentNode) {
```

Comment:
1) addViewer() method should be called anywhere. Even should be created and destroyed dynamically on runtime.
2) The interfaces (methods) of Workbench Part should be independent from specific widget tool kit (like dojo).